Build failure: notepad-next

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

```
❯ nix run nixpkgs#notepad-next                                                                
error: Cannot build '/nix/store/im1lql1c0byn58rrpqhaalscqjv5j5j6-notepad-next-0.12.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/45bz24s8rdc4prxd5w4bicyphigq2jlp-notepad-next-0.12
       Last 25 log lines:
       > In file included from ../src/scintilla/qt/ScintillaEdit/ScintillaDocument.cpp:24:
       > ../src/scintilla/include/Scintilla.h:32:19: note: 'uptr_t' declared here
       >    32 | typedef uintptr_t uptr_t;
       >       |                   ^~~~~~
       > ../src/scintilla/src/ViewStyle.h:42:75: error: 'sptr_t' is not a member of 'Scintilla'; did you mean 'sptr_t'?
       >    42 | inline ColourOptional OptionalColour(Scintilla::uptr_t wParam, Scintilla::sptr_t lParam) noexcept {
       >       |                                                                           ^~~~~~
       > ../src/scintilla/include/Scintilla.h:34:18: note: 'sptr_t' declared here
       >    34 | typedef intptr_t sptr_t;
       >       |                  ^~~~~~
       > ../src/scintilla/src/ViewStyle.h:42:90: error: expected ',' or ';' before 'noexcept'
       >    42 | inline ColourOptional OptionalColour(Scintilla::uptr_t wParam, Scintilla::sptr_t lParam) noexcept {
       >       |                                                                                          ^~~~~~~~
       > g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -D_REENTRANT -fPIC -DAPP_VERSION="\"0.12\"" -DAPP_COPYRIGHT="\"Copyright 2019-2025 Justin Dailey\"" -DAPP_DISTRIBUTION="\"\"" -DSCI_OWNREGEX -DADS_STATIC -DQAPPLICATION_CLASS=QApplication -DSCINTILLA_QT=1 -DMAKING_LIBRARY=1 -DSCI_LEXER=1 -D_CRT_SECURE_NO_DEPRECATE=1 -DEXPORT_IMPORT_API= -DNDEBUG=1 -DLUA_PATH_DEFAULT=\"\" -DLUA_CPATH_DEFAULT=\"\" -DLUA_VERSION_NUM=503 -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src/NotepadNext -I. -I../src/singleapplication -I../src/scintilla/qt/ScintillaEditBase -I../src/scintilla/qt/ScintillaEdit -I../src/scintilla/include -I../src/scintilla/src -I../src/lexilla/access -I../src/lexilla/lexlib -I../src/lexilla/include -I../src/uchardet/src -I../src/lua/src -I../src/ads/src -I../src/editorconfig-core-qt -I../src/NotepadNext/decorators -I../src/NotepadNext/dialogs -I../src/NotepadNext/docks -I../src/NotepadNext/widgets -I../src -I../src/lexilla/include -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtPrintSupport -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtWidgets -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui/5.15.18 -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui/5.15.18/QtGui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtNetwork -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore/5.15.18 -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore/5.15.18/QtCore -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore -Ibuild/moc -I/nix/store/yydc0n1z5ag31x2vqv05n8iz498nvlm2-libglvnd-1.7.0-dev/include -Ibuild/ui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/mkspecs/linux-g++ -o build/obj/KeyMap.o ../src/scintilla/src/KeyMap.cxx
       > ../src/scintilla/src/ViewStyle.h:229:67: error: 'Scintilla::uptr_t' has not been declared
       >   229 |         bool SetElementColourOptional(Scintilla::Element element, Scintilla::uptr_t wParam, Scintilla::sptr_t lParam);
       >       |                                                                   ^~~~~~~~~
       > ../src/scintilla/src/ViewStyle.h:229:93: error: 'Scintilla::sptr_t' has not been declared
       >   229 |         bool SetElementColourOptional(Scintilla::Element element, Scintilla::uptr_t wParam, Scintilla::sptr_t lParam);
       >       |                                                                                             ^~~~~~~~~
       > g++ -c -pipe -O2 -std=gnu++1z -Wall -Wextra -D_REENTRANT -fPIC -DAPP_VERSION="\"0.12\"" -DAPP_COPYRIGHT="\"Copyright 2019-2025 Justin Dailey\"" -DAPP_DISTRIBUTION="\"\"" -DSCI_OWNREGEX -DADS_STATIC -DQAPPLICATION_CLASS=QApplication -DSCINTILLA_QT=1 -DMAKING_LIBRARY=1 -DSCI_LEXER=1 -D_CRT_SECURE_NO_DEPRECATE=1 -DEXPORT_IMPORT_API= -DNDEBUG=1 -DLUA_PATH_DEFAULT=\"\" -DLUA_CPATH_DEFAULT=\"\" -DLUA_VERSION_NUM=503 -DQT_NO_DEBUG -DQT_PRINTSUPPORT_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_CORE_LIB -I../src/NotepadNext -I. -I../src/singleapplication -I../src/scintilla/qt/ScintillaEditBase -I../src/scintilla/qt/ScintillaEdit -I../src/scintilla/include -I../src/scintilla/src -I../src/lexilla/access -I../src/lexilla/lexlib -I../src/lexilla/include -I../src/uchardet/src -I../src/lua/src -I../src/ads/src -I../src/editorconfig-core-qt -I../src/NotepadNext/decorators -I../src/NotepadNext/dialogs -I../src/NotepadNext/docks -I../src/NotepadNext/widgets -I../src -I../src/lexilla/include -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtPrintSupport -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtWidgets -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui/5.15.18 -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui/5.15.18/QtGui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtGui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtNetwork -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore/5.15.18 -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore/5.15.18/QtCore -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/include/QtCore -Ibuild/moc -I/nix/store/yydc0n1z5ag31x2vqv05n8iz498nvlm2-libglvnd-1.7.0-dev/include -Ibuild/ui -I/nix/store/cm0w4zim0r53dmilrr5v5yas49801zq0-qtbase-5.15.18-dev/mkspecs/linux-g++ -o build/obj/Indicator.o ../src/scintilla/src/Indicator.cxx
       > make[1]: *** [Makefile:3305: build/obj/ScintillaDocument.o] Error 1
       > make[1]: *** Waiting for unfinished jobs....
       > make[1]: Leaving directory '/build/source/NotepadNext'
       > make: *** [Makefile:47: sub-NotepadNext-make_first] Error 2
       For full logs, run:
         nix log /nix/store/im1lql1c0byn58rrpqhaalscqjv5j5j6-notepad-next-0.12.drv

```

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

hydra search doesn;t work for me

### Relevant log output

```console
see above
```

### Additional context

https://github.com/NixOS/nixpkgs/issues/475479

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.3-cachyos, NixOS, 26.05 (Yarara), 26.05.20260107.5912c17`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+2`
 - nixpkgs: `/nix/store/i77jvi0qh4kddj4hraizq1j0c028q05l-source`


### Notify maintainers

@SebTM


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/no/notepad-next/package.nix
+++ b/pkgs/by-name/no/notepad-next/package.nix
@@ -25,6 +25,12 @@ stdenv.mkDerivation (finalAttrs: {
   ];
   buildInputs = [ qt5.qtx11extras ];
 
+  # Fix build with GCC 14+: Scintilla needs cstdint for intptr_t/uintptr_t types
+  # https://github.com/dail8859/NotepadNext/issues/752
+  postPatch = ''
+    sed -i '1i #include <cstdint>' src/scintilla/include/ScintillaTypes.h
+  '';
+
   qmakeFlags = [
     "PREFIX=${placeholder "out"}"
     "src/NotepadNext.pro"

